components/insight/SRC/org/openmicroscopy/shoola/env/data/DataServicesFactory.java

```diff
@@ -634,10 +633,12 @@ public void connect(UserCredentials uc)
         Set<GroupData> available;
         List<ExperimenterData> exps = new ArrayList<ExperimenterData>();
         String ldap = null;
+        long gid = exp.getDefaultGroup().getId();
+        SecurityContext ctx = new SecurityContext(gid);
+        boolean canCreate = omeroGateway.canCreate(ctx);
```

Comment:
Do you mean `Boolean.TRUE.equals(omeroGateway.canCreate(ctx))` to guard against nulls?

Reply:
Ah, sorry, maybe not, probably getting my gateways mixed up!

Reply:
@jburel : if we start to take more conditions into account, how will this change in the future? Will you break the API?

Reply:
This internal gateway is now in many cases a wrapper around the Java gateway under blitz.
If we introduce more conditions, I think we will be adding entry in the lookup. It won't break the API.